restore/restore.go

```diff
@@ -96,18 +98,71 @@ func GetResponse(client *http.Client, req *http.Request) *http.Response {
 	return resp
 }
 
-func (c *RestoreCommand) StartRestore(cliConnection plugin.CliConnection, serviceInstanceName string, backupId string) {
+func (c *RestoreCommand) StartRestore(cliConnection plugin.CliConnection, serviceInstanceName string, backupId string, timeStamp string, isGuidOperation bool, instanceGuidOperation bool, instanceNameOperation bool, instanceGuid string, instanceName string, deletedFlag bool) {
 	fmt.Println("Starting restore for ", AddColor(serviceInstanceName, cyan), "...")
-
+	var source_instance_id string = ""
+	if instanceGuidOperation == true {
+		var serviceInstanceName string = ""
+		serviceInstanceName = guidTranslator.FindInstanceName(cliConnection, instanceGuid, nil)
+		fmt.Println("Instance Name = ", serviceInstanceName)
+		if serviceInstanceName == "" {
+			fmt.Println(AddColor("FAILED", red))
+			fmt.Println("Error - No service instance found with service instance guid = ", instanceGuid)
+			os.Exit(7)
+		} else {
+			source_instance_id = instanceGuid
+		}
+	}
+	if instanceNameOperation == true {
+		guid := guidTranslator.FindInstanceGuid(cliConnection, instanceName, nil, "")
+		guid = strings.Trim(guid, ",")
+		guid = strings.Trim(guid, "\"")
+		source_instance_id = guid
+	}
+	if deletedFlag == true {
+		var guidMap map[string]string = guidTranslator.FindDeletedInstanceGuid(cliConnection, instanceName, nil, "")
+		if len(guidMap) > 1 {
+			fmt.Println(AddColor("FAILED", constants.Red))
+			fmt.Println("" + instanceName + " maps to multiple instance GUIDs, please use 'cf instance-events --delete' to list all instance delete events, get required instance guid from the list and then use 'cf list-backup --guid GUID' to fetch backups list.")
+			fmt.Println("Enter 'cf backup' to check the list of commands and their usage.")
+			os.Exit(1)
+		} else {
+			for k, _ := range guidMap {
+				guid := k
+				guid = strings.Trim(guid, ",")
```

Comment:
same as above .. why the trim's are required ?

also if required , can you extract this to a util function and reuse it ?